### DISCUSSION TITLE
Dev server serving files from "dist" instead of changed ones

### DISCUSSION THREAD

**Discussion by F30** [parcel-bundler/parcel#7762] [Q&A]
I experienced some similar problems as described in #1591, i.e. hot reloading not working with `parcel serve` in Parcel 2.3.2. In fact, no code reloading (even with restarting the dev server) worked until I ran `parcel build`.

I tracked it down to the following behavior:

Everything works fine as long as no "dist" folder is present. As soon as `parcel build` has been executed once and the "dist" folder is present, files from there appear to take precedence. This means the dev server will not serve changed files until running another build (or deleting the "dist" folder, for that matter). Notably, the dev server does detect changes and issue a rebuild, but the changes do not get served.

Is this the expected behavior? I find it rather confusing and couldn't find any description of it, though it's hard to search for it. If so, what is the suggested way to avoid it? Configure separate targets for dev and prod, and never running `parcel build` for the dev target?

Thanks for your support!

**Comment by F30** [comment]
>  Configure separate targets for dev and prod, and never running `parcel build` for the dev target?

I played around with separate targets a bit.

As it turns out, `parcel serve` always populates the "dist" directory. It does that even if a different `distDir` has been configured for a target and that target gets specified using `--target`.

However, `parcel build` does respect the `distDir` setting. This means one can work around the issue by using a different dist directory for the build output:

```
"targets": {
  "default": {
    "distDir": "dist-build"
  }
}
```

**Comment by nazreen** [comment]
TLDR for me and others, quick fix: delete the `dist` folder then run to serve again
That fixed it for me

**Comment by Hongjia-Liu** [comment]
If Hot Module Reloading is not working, the problem may relate to the periods (`.`) of the directories name. We can try to get rid of them.

Consider changing the `import` statement from relative path to absolute path, and then changeing it back. This magically works for me.
```jsx
import App from "./App";
import App from "/src/App"
```
